src/cli/rustup_mode.rs

```diff
@@ -806,14 +805,7 @@ async fn check_updates(cfg: &Cfg<'_>, opts: CheckOpts) -> Result<utils::ExitCode
     // See: https://github.com/rust-lang/futures-rs/pull/1194#discussion_r209501774
     if num_channels > 0 {
         let multi_progress_bars =
-            MultiProgress::with_draw_target(match cfg.process.var("RUSTUP_TERM_PROGRESS_WHEN") {
-                Ok(s) if s.eq_ignore_ascii_case("always") => {
-                    ProgressDrawTarget::term_like(Box::new(t))
-                }
-                Ok(s) if s.eq_ignore_ascii_case("never") => ProgressDrawTarget::hidden(),
-                _ if is_a_tty => ProgressDrawTarget::term_like(Box::new(t)),
-                _ => ProgressDrawTarget::hidden(),
-            });
+            MultiProgress::with_draw_target(cfg.process.progress_draw_target());
```

Comment:
When extracting existing logic, the recommended approach would be doing the extraction first in a separate commit, and then reuse the newly-extracted function in your new code. Thus, the order of the 2 current commits should be the other way around.